src/Databases/DatabaseReplicated.cpp

```diff
@@ -1083,12 +1083,14 @@ void DatabaseReplicated::recoverLostReplica(const ZooKeeperPtr & current_zookeep
     }
     LOG_INFO(log, "All tables are created successfully");
 
-    if (max_log_ptr_at_creation != 0)
+    chassert(max_log_ptr_at_creation || our_log_ptr);
```

Comment:
@tavplubix sorry for the dumb question, but if replica is died between database being created and this line, next time when the replica is up, it will always be stuck on this line right? Since `max_log_ptr_at_creation` is always 0 and `our_log_ptr` is also 0. I seem to run into this situation when trying to make some code change, but I feel it is also possible to happen in reality?

Reply:
It's a good question, you have found a bug! Yes, debug builds and builds with sanitizers will always crash on this line with SIGABRT. Release builds ignore assertions, so it will continue executing this code and will not create the nodes in zk. As a result, some DDL queries may throw `Watching task {} is executing longer than distributed_ddl_task_timeout ...` (if they started while the replica was being created)